src/ng/controller.js

```diff
@@ -1,5 +1,15 @@
 'use strict';
 
+var CNTRL_REG = /^(\S+)(\s+as\s+(\w+))?$/;
+function identifierForController(controller, ident) {
+  if (ident && isString(ident)) return ident;
+  if (isString(controller)) {
+    var match = CNTRL_REG.exec(controller);
+    if (match) return match[3];
+  }
+}
```

Comment:
In that case this function needs refactoring to return the controller name as well, since that is what is needed in $controller.

All the references to `label`, `ident` and `identifier` in this file ought to be changed to `alias` for consistency.  If you don't like `alias` then you can choose an alternate and change all the references to `ident`, `identifier`, `alias` and `label` to a single consistent term throughout the codebase.